File: wk01/ps/mario-less/mario.c
#include <cs50.h>
#include <stdio.h>

int get_height(void);
void print_pyramid(int);

int main(void)
{
    int n = get_height();
    print_pyramid(n);
}

int get_height(void)
{
    int n;
    do
    {
        n = get_int("Height: ");
    }
    while (n < 1 || n > 8);
    return n;
}

void print_pyramid(int n)
{
    for (int i = 0; i < n; i++)
    {
        for (int j = 0; j < (n - (i + 1)); j++)
        {
            printf(" ");  // Print spaces one less than height
        }

        for (int j = 0; j < i + 1; j++)
        {
            printf("#");  // Print #s equal to row number
        }

        printf("\n");
    }

}

File: wk06/ps/sentimental-mario-less/mario.py
from cs50 import get_int


def main():
    n = get_height()
    print_pyramid(n)


def get_height():
    while True:
        n = get_int("Height: ")
        if n >= 1 and n <= 8:
            return n
        else:
            print("Height should be between 1 and 8 inclusive. Please re-enter.")


def print_pyramid(n):
    for i in range(1, n + 1):
        spaces = ' ' * (n - i)
        blocks = '#' * i
        print(spaces + blocks)


if __name__ == "__main__":
    main()